editor/src/components/custom-code/code-file.ts

```diff
@@ -570,6 +572,38 @@ export function normalisePathToUnderlyingTargetForced(
   )
 }
 
+export function findUnderlyingTargetComponentImplementationFromImportInfo(
+  projectContents: ProjectContentTreeRoot,
+  importInfo: ImportInfo | null,
+): UtopiaJSXComponent | null {
+  if (importInfo == null) {
+    return null
+  }
+
+  const projectContentsFlattened = treeToContents(projectContents)
+
+  const variableName =
+    importInfo.type === 'SAME_FILE_ORIGIN' ? importInfo.variableName : importInfo.exportedName
+
+  // we have to find the element based on the top level name
+  const file = projectContentsFlattened[importInfo.filePath]
```

Comment:
```suggestion
  const file = getContentsTreeFileFromString(projectContents, importInfo.filePath)
```

Then you don't need to invoke `treeToContents`.

Reply:
ah I actually wanted to ask this!!

Reply:
thanks, I fixed it